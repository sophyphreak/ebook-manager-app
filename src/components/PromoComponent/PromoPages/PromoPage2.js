import React, { Component } from 'react';
import {
  Card,
  CardHeader,
  CardBody,
  Button,
  Form
} from 'reactstrap';

import Price from '../../FormElements/Price';
import PromoType from '../../FormElements/PromoType';
import DatesPicker from '../../FormElements/DatesPicker';

const PromoPage2 = ({
  error,
  onBack,
  onSubmit,

  price,
  onPriceChange,

  promoType,
  onPromoTypeChange,

  startDate,
  endDate,
  focusedInput,
  onDatesChange,
  onFocusChange
}) => (
    <Card className="animated fadeIn">
      <CardHeader>
        <h4><strong>Book Promo - 2 of 3</strong></h4>
      </CardHeader>
      <CardBody>
        <Form onSubmit={onSubmit}>
          <Price
            value={price}
            onChange={onPriceChange}
          />
          <PromoType
            value={promoType}
            onChange={onPromoTypeChange}
          />
          <DatesPicker
            startDate={startDate}
            endDate={endDate}
            onDatesChange={onDatesChange}
            focusedInput={focusedInput}
            onFocusChange={onFocusChange}
          />
          {error && <p>{error}</p>}
          <Button
            onClick={onBack}
            size="sm"
            color="basic"
          >
            Go Back
          </Button>
          <Button
            type="submit"
            size="sm"
            color="primary"
            value="Submit"
          >
            Next Section
          </Button>
        </Form>
      </CardBody>
    </Card>
  );

export default PromoPage2;