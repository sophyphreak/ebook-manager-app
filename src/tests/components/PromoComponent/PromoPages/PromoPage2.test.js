import React from 'react';
import { shallow } from 'enzyme';
import PromoPage2 from '../../../../components/PromoComponent/PromoPages/PromoPage2';

let error, wrapper;

beforeEach(() => {
  error = {
    message: "",
    price: "",
    promoType: ""
  };
  wrapper = shallow(<PromoPage2 error={error} />);
});

test('should render PromoPage2 correctly', () => {
  expect(wrapper).toMatchSnapshot();
});