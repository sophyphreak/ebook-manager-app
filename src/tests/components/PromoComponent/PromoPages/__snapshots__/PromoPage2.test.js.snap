// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PromoPage2 correctly 1`] = `
<Card
  className="animated fadeIn"
  tag="div"
>
  <CardHeader
    tag="div"
  >
    <h4>
      <strong>
        Book Promo - 2 of 3
      </strong>
    </h4>
  </CardHeader>
  <CardBody
    tag="div"
  >
    <Form
      tag="form"
    >
      <Price
        errorMessage=""
        hasError={false}
      />
      <PromoTypes
        errorMessage=""
        hasError={false}
      />
      <DatesPicker />
      <Button
        className="back-button"
        color="basic"
        size="sm"
        tag="button"
      >
        Go Back
      </Button>
      <Button
        color="primary"
        size="sm"
        tag="button"
        type="submit"
        value="Submit"
      >
        Next Section
      </Button>
    </Form>
  </CardBody>
</Card>
`;
